dotCMS/src/integration-test/java/com/dotcms/security/secret/ServiceIntegrationAPIImplTest.java

```diff
@@ -0,0 +1,302 @@
+        api.saveSecrets(bean3Host1, host1, admin);
+        api.saveSecrets(bean1Host2, host2, admin);
+
+        final Map<String,List<String>> serviceKeysByHost = api.serviceKeysByHost();
+        assertNotNull(serviceKeysByHost.get(host1.getIdentifier()));
+        assertNotNull(serviceKeysByHost.get(host2.getIdentifier()));
+
+        assertEquals(3, api.listServiceKeys(host1, admin).size());
+        assertEquals(1, api.listServiceKeys(host2, admin).size());
+
+        assertEquals(3, serviceKeysByHost.get(host1.getIdentifier()).size());
+        assertEquals(1, serviceKeysByHost.get(host2.getIdentifier()).size());
+
+        assertTrue(serviceKeysByHost.get(host1.getIdentifier()).contains("serviceKey-1-Host-1".toLowerCase()));
+        assertTrue(serviceKeysByHost.get(host1.getIdentifier()).contains("serviceKey-2-Host-1".toLowerCase()));
+        assertTrue(serviceKeysByHost.get(host1.getIdentifier()).contains("serviceKey-3-Host-1".toLowerCase()));
+        assertTrue(serviceKeysByHost.get(host2.getIdentifier()).contains("serviceKey-1-Host-2".toLowerCase()));
+
+    }
+
+    @Test
+    public void Test_Lookup_Fallback() throws DotDataException, DotSecurityException {
+        final ServiceIntegrationAPI api = APILocator.getServiceIntegrationAPI();
+        final User admin = TestUserUtils.getAdminUser();
+        final Host systemHost = APILocator.systemHost();
+        final Host host1 = new SiteDataGen().nextPersisted();
+
+        final ServiceSecrets.Builder builder1 = new ServiceSecrets.Builder();
+
+        final ServiceSecrets beanSystemHost1 = builder1.withServiceKey("serviceKey-1-Any-Host")
+                .withHiddenSecret("fallback:test:secret1", "sec1")
+                .withSecret("fallback:test", true)
+                .build();
+
+        api.saveSecrets(beanSystemHost1, systemHost, admin);
+
+        final Optional<ServiceSecrets> serviceSecretsOptional = api.getSecretsForService("serviceKey-1-Any-Host", host1, admin);
+        assertTrue(serviceSecretsOptional.isPresent());
+        final ServiceSecrets recoveredBean = serviceSecretsOptional.get();
+        assertEquals("serviceKey-1-Any-Host", recoveredBean.getServiceKey());
+        assertTrue(recoveredBean.getSecrets().containsKey("fallback:test"));
+        assertTrue(recoveredBean.getSecrets().containsKey("fallback:test:secret1"));
+    }
+
+    @Test
+    public void Test_Save_Secrets_Then_Replace_Secret() throws DotDataException, DotSecurityException {
+        final ServiceIntegrationAPI api = APILocator.getServiceIntegrationAPI();
+        final User admin = TestUserUtils.getAdminUser();
+        final Host host = new SiteDataGen().nextPersisted();
+
+        final ServiceSecrets.Builder builder1 = new ServiceSecrets.Builder();
+        final ServiceSecrets secrets1 = builder1.withServiceKey("serviceKey-1-Host-1")
+                .withHiddenSecret("test:secret1", "sec1")
+                .withHiddenSecret("test:secret2", "sec2")
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "test:secret2" appears 10 times in this file; the first occurrence is on line 173](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4623512)